pkg/restore/prioritize_group_version_test.go

```diff
@@ -0,0 +1,367 @@
+	}{
+		{
+			name: "sorts Kubernetes API group versions per k8s priority",
+			orig: []metav1.GroupVersionForDiscovery{
+				{Version: "v2"},
+				{Version: "v11alpha2"},
+				{Version: "foo10"},
+				{Version: "v10"},
+				{Version: "v12alpha1"},
+				{Version: "v3beta1"},
+				{Version: "foo1"},
+				{Version: "v1"},
+				{Version: "v10beta3"},
+				{Version: "v11beta2"},
+			},
+			want: []metav1.GroupVersionForDiscovery{
+				{Version: "v10"},
+				{Version: "v2"},
+				{Version: "v1"},
+				{Version: "v11beta2"},
+				{Version: "v10beta3"},
+				{Version: "v3beta1"},
+				{Version: "v12alpha1"},
+				{Version: "v11alpha2"},
+				{Version: "foo1"},
+				{Version: "foo10"},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			k8sPrioritySort(tc.orig)
+
+			assert.Equal(t, tc.want, tc.orig)
+		})
+	}
+}
+
+func TestUserResourceGroupVersionPriorities(t *testing.T) {
+	tests := []struct {
+		name string
+		cm   *corev1.ConfigMap
+		want map[string]metav1.APIGroup
+	}{
+		{
+			name: "retrieve version priority data from config map",
+			cm: builder.
+				ForConfigMap("velero", "enableapigroupversions").
+				Data(
+					"restoreResourcesVersionPriority",
+					`rockbands.music.example.io=v2beta1,v2beta2
+orchestras.music.example.io=v2,v3alpha1
+subscriptions.operators.coreos.com=v2,v1`,
+				).
+				Result(),
+			want: map[string]metav1.APIGroup{
+				"rockbands.music.example.io": {Versions: []metav1.GroupVersionForDiscovery{
+					{Version: "v2beta1"},
+					{Version: "v2beta2"},
+				}},
+				"orchestras.music.example.io": {Versions: []metav1.GroupVersionForDiscovery{
+					{Version: "v2"},
+					{Version: "v3alpha1"},
+				}},
+				"subscriptions.operators.coreos.com": {Versions: []metav1.GroupVersionForDiscovery{
+					{Version: "v2"},
+					{Version: "v1"},
+				}},
+			},
+		},
+		{
+			name: "incorrect data format returns no user version priorities",
+			cm: builder.
+				ForConfigMap("velero", "enableapigroupversions").
+				Data(
+					"restoreResourcesVersionPriority",
+					`rockbands.music.example.io=v2beta1,v2beta2\n orchestras.music.example.io=v2,v3alpha1`,
+				).
+				Result(),
+			want: nil,
+		},
+		{
+			name: "incorrect and correct data format returns some user version priorities",
+			cm: builder.
+				ForConfigMap("velero", "enableapigroupversions").
+				Data(
+					"restoreResourcesVersionPriority",
+					`rb=foo,foo2
+o=v2beta2
+\n`,
+				).
+				Result(),
+			want: map[string]metav1.APIGroup{
+				"rb": {Versions: []metav1.GroupVersionForDiscovery{
+					{Version: "foo"},
+					{Version: "foo2"},
+				}},
+				"o": {Versions: []metav1.GroupVersionForDiscovery{
+					{Version: "v2beta2"},
+				}},
+			},
+		},
+	}
+
+	fakeCtx := &restoreContext{
+		log: test.NewLogger(),
+	}
+
+	for _, tc := range tests {
+		t.Log(tc.name)
+		priorities := userResourceGroupVersionPriorities(fakeCtx, tc.cm)
+		assert.Equal(t, tc.want, priorities)
+	}
+}
+
+func TestFindTargetGroup(t *testing.T) {
```

Comment:
This test should be renamed as it's the function `findAPIGroup` that's under test here.

Reply:
Good catch. I'll have that changed.